src/Xamarin.Android.Build.Tasks/Xamarin.Android.Legacy.targets

```diff
@@ -376,14 +376,14 @@ projects. .NET 5 projects will not import this file.
       <_ResolvedAssemblies          Include="@(ResolvedAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(Filename)%(Extension)')" />
       <_ResolvedSymbols             Include="@(ResolvedSymbols->'$(MonoAndroidIntermediateAssemblyDir)%(Filename)%(Extension)')" />
       <_ResolvedUserAssemblies      Include="@(ResolvedUserAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(Filename)%(Extension)')" />
-      <_ResolvedFrameworkAssemblies Include="@(ResolvedFrameworkAssemblies)" />
+      <_ResolvedFrameworkAssemblies Include="@(ResolvedFrameworkAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(Filename)%(Extension)')" />
     </ItemGroup>
-    <ItemGroup Condition=" ('$(AndroidLinkMode)' == 'None' Or '$(AndroidUseSharedRuntime)' == 'true') ">
+    <ItemGroup Condition=" ('$(AndroidLinkMode)' == 'None' Or '$(Optimize)' != 'true') ">
       <_ShrunkAssemblies          Include="@(_ResolvedAssemblies)" />
       <_ShrunkUserAssemblies      Include="@(_ResolvedUserAssemblies)" />
       <_ShrunkFrameworkAssemblies Include="@(_ResolvedFrameworkAssemblies)" />
     </ItemGroup>
-    <ItemGroup Condition=" '$(AndroidLinkMode)' != 'None' And '$(AndroidUseSharedRuntime)' != 'true' ">
+    <ItemGroup Condition=" '$(AndroidLinkMode)' != 'None' And '$(Optimize)' == 'true' ">
```

Comment:
I think these need to use `$(AndroidIncludeDebugSymbols)` instead of `$(Optimize)`.

The logic in these three places needs to all be the equivalent:

https://github.com/xamarin/xamarin-android/blob/398bf411a4e010b63e8c14f8c2bd1422559f168b/src/Xamarin.Android.Build.Tasks/Tasks/ProcessAssemblies.cs#L103

https://github.com/xamarin/xamarin-android/blob/398bf411a4e010b63e8c14f8c2bd1422559f168b/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1835

https://github.com/xamarin/xamarin-android/blob/398bf411a4e010b63e8c14f8c2bd1422559f168b/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.AssemblyResolution.targets#L112-L121

Note, that I don't think master is quite right at the moment...